### ISSUE TITLE
md-menu

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#119]
Blocked on both #113

**Comment by jelbourn** [comment]
First step for alpha.6: 
- [ ] design doc
- [ ] proof-of-concept w/ a11y

**Comment by fxck** [comment]
@kara can you post the design doc here once it's done, please?

**Comment by kara** [comment]
https://docs.google.com/document/d/1cSUrY2WAqSN8ePldLLqrEVkk2BkYbPB2fysC6brvj4Q/preview

**Comment by willfarrell** [comment]
Noticed `<md-menu-divider></md-menu-divider>` was not in the design doc. Are there plans to include it?

**Comment by quanterion** [comment]
Also what about plans to include check boxes inside menu?

**Comment by riankol** [comment]
I noticed md-menu is not longer an 'Initial version', but the 'custom off-set support' is not implemented yet. Can you give me any indication when this is expected to be implemented?
Also I would like to use a `md-menu-divider`. This is not in the design doc, like willfarrell mentioned above. Did you already decide if (and if so, when) a `md-menu-divider` will be implemented?

**Comment by jelbourn** [comment]
@riankol can you file a separate feature request for that?

@quanterion the menu isn't meant for item selection, but for performing actions. `md-select` will be used for selection. 

I'm closing this issue since most of the core menu behavior is done; remaining features can be tracked via their own issues.

**Comment by mhmo91** [comment]
@jelbourn What if i want the menu to be clicked open on hover? are u planning such a feature? It's really important for desktop UX. 
thanks

**Comment by fxck** [comment]
it's clunky, but I think you could do this 

```
<button [md-menu-trigger-for]="menu" #menuTrigger="mdMenuTrigger" (mouseenter)="menuTrigger.openMenu()">
   <md-icon>more_vert</md-icon>
</button>
```

**Comment by mhmo91** [comment]
@fxck  it didn't work, n threw an error " self._MdMenu_21_3.context.openMenu is not a function ..."

**Comment by mhmo91** [comment]
i have been working with angular for sometime now, i'm loving it's design patterns, but it's biggest downside is the lack of ui components... even the one available doesn't have some of the basic functionality we r used to have... n this is freaking time consuming!

**Comment by SamiAl90** [comment]
You'd be helped reading the docs so you know how to trigger the menu manually:
https://github.com/angular/material2/tree/master/src/lib/menu

**Comment by teterovic** [comment]
Any plans on implementing cascading menus, is it on the roadmap ?

**Comment by lnardai** [comment]
In this document(linked above in this thread), at the end, they are mentioning a TODO, and there is this entry:

> Roadmap and timeline
> The basic functionality should be completed in time for alpha.7.  Extras like "offset" may be deferred until later.
> 
> TODO:
> Cascading menu support

[https://docs.google.com/document/d/1cSUrY2WAqSN8ePldLLqrEVkk2BkYbPB2fysC6brvj4Q/preview](https://docs.google.com/document/d/1cSUrY2WAqSN8ePldLLqrEVkk2BkYbPB2fysC6brvj4Q/preview)

Sadly, this is a blocker for us in switching to material2.

**Comment by jelbourn** [comment]
@crisbeto is starting work on the nested menus this week

**Comment by clement-hollander** [comment]
Any update on this ?

**Comment by jelbourn** [comment]
https://github.com/angular/material2/pull/5493
